presto-main/src/main/java/com/facebook/presto/operator/StandardJoinFilterFunctionVerifier.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.Block;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class StandardJoinFilterFunctionVerifier
+        implements JoinFilterFunctionVerifier
+{
+    private static final Block[] EMPTY_BLOCK_ARRAY = new Block[0];
+
+    private final JoinFilterFunction filterFunction;
+    private final List<Block[]> channelArrays;
+
+    public StandardJoinFilterFunctionVerifier(JoinFilterFunction filterFunction, List<List<Block>> channels)
+    {
+        this.channelArrays = buildChannelArrays(requireNonNull(channels, "channels can not be null"));
+        this.filterFunction = requireNonNull(filterFunction, "filterFunction can not be null");
+    }
+
+    private List<Block[]> buildChannelArrays(List<List<Block>> channels)
```

Comment:
I find this code quite confusing.  I think it is taking a column oriented data set (i.e., a list of columns each of which is a list of blocks) and converting it into a page oriented data (i.e., a list of pages each of which is a list of column-blocks), but the name and variables seem to indicate the conversion is in the other direction.

As mentioned above, I would inline this into the constructor, and then I would rename as follows:
`channelsArray` -> `pages` (both variable and field)
`i` -> `pageIndex`
`j` -> `channelIndex`